modules: mcuboot: Disable Zephyr's MCUboot handling for child images

MarekPieta opened the pull request:
Change disables Zephyr's MCUboot handling in child images. The NCS handles needed operations on its own. Change is needed to ensure consistency for main application image and child images.

Jira: NCSDK-22347

Commit:
manifest: Update sdk-trusted-firmware-m revision

Pull alignment of UARTE CMSIS driver to nrfx 3.0.0

Signed-off-by: Adam Wojasinski <adam.wojasinski@nordicsemi.no>

Commit:
dm: time: Replace depricated RTC symbols to existing in nrfx3.0

New nrfx release added namespace prefix to RTC symbols. This patch
aligns code to renamed symbols.

Signed-off-by: Adam Wojasinski <adam.wojasinski@nordicsemi.no>

Commit:
applications: nrf5340_audio: Align to multi instance I2S driver

nrfx 3.0.0 introduces multi instance support for I2S driver.
Aligned I2S driver API used in application and Kconfig symbols
to enable instance 0.

Signed-off-by: Adam Wojasinski <adam.wojasinski@nordicsemi.no>

Commit:
debug: cpu_load: Align TIMER configuration macro usage

Configuration macro for TIMER takes as a parameter frequency of the
timer that user wants to set. This patch align macro call to that.

Signed-off-by: Adam Wojasinski <adam.wojasinski@nordicsemi.no>

Commit:
samples: peripheral: 802154_phy_test: Align GPIOTE and TIMER APIs

GPIOTE driver has been complitely reworked in nrfx 3.0.0.
Aligned code in a way that is suggested in migration notes from
nrfx 2.11.0 to nrfx 3.0.0 that is available at
https://github.com/NordicSemiconductor/nrfx/wiki/nrfx-2.11.0-to-3.0.0

Signed-off-by: Adam Wojasinski <adam.wojasinski@nordicsemi.no>

MarekPieta committed:
applications: nrf_desktop: Disable image list hash support on nRF52820

Change disables CONFIG_BOOT_SERIAL_IMG_GRP_HASH in MCUboot bootloader
configuration of boards that use nRF52820 SoC. This is needed to reduce
the memory consumption.

Jira: NCSDK-21638

Signed-off-by: Marek Pieta <Marek.Pieta@nordicsemi.no>

gopiotr committed:
scripts: quarantine: add tests with old Ztest API

Since Zephyr 3.4 old Ztest API is deprecated and all tests with old
API fail during building by Twister.

Signed-off-by: Piotr Golyzniak <piotr.golyzniak@nordicsemi.no>

rlubos committed:
lib: date_time: Fix dependency to POSIX network APIs

SNTP library no longer relies on POSIX network APIs/headers, and there's
no need for date_time lib to rely on them either. Therefore, make the
library indifferent on POSIX configuration by using zsock APIs directly.

Signed-off-by: Robert Lubos <robert.lubos@nordicsemi.no>

rlubos committed:
zigbee: Align with upstream 15.4 API changes

Align zigbee codebase with upstream 15.4 API changes.

Signed-off-by: Robert Lubos <robert.lubos@nordicsemi.no>

rlubos committed:
mgmt: mcumgr: Align codebase with the new callback format

Align NCS codebase with the new struct mgmt_callback format.

Signed-off-by: Robert Lubos <robert.lubos@nordicsemi.no>

rlubos committed:
samples: nrf9160: lwm2m_client: Prevent name collission with sensor API

sensor_read() is now a part of the public sensor AI, therefore need to
use a different name in the sample to avoid conflict.

Signed-off-by: Robert Lubos <robert.lubos@nordicsemi.no>

rlubos committed:
lib: st25r3911b: Align with upstream SPI CS changes

Align with the upstream SPI CS API changes.

Signed-off-by: Robert Lubos <robert.lubos@nordicsemi.no>

rlubos committed:
bluetooth: mesh: Align with the upstream API changes

Align bluetooth mesh tests with the upstream API changes.

Signed-off-by: Robert Lubos <robert.lubos@nordicsemi.no>

rlubos committed:
libc: Avoid using deprecated MINIMAL_LIBC_MALLOC_ARENA_SIZE

CONFIG_MINIMAL_LIBC_MALLOC_ARENA_SIZE was deprecated and should no
longer be used.

Signed-off-by: Robert Lubos <robert.lubos@nordicsemi.no>

maje-emb committed:
samples: bluetooth: peripheral_mds: Disable `SENSOR_SHELL`

To reduce RAM usage, `SENSOR_SHELL` is disabled.

Signed-off-by: Marcin Jelinski <marcin.jelinski@nordicsemi.no>

aasinclair committed:
samples: pmic: npm1300_fuel_gauge: Added Kconfig modules

Required modules have been added to prj.conf.
These were previously added by the shield defconfig but
this has been deleted upstream.

Signed-off-by: Andy Sinclair <andy.sinclair@nordicsemi.no>

rlubos committed:
samples: bluetooth: mesh: silvair_enocean: Fix Kconfig dependencies

CONFIG_BT_HOST_CCM, which is a dependency for CONFIG_BT_ENOCEAN is no
longer enabled automatically by Zephyr, hence enable it manually in the
project file.

Signed-off-by: Robert Lubos <robert.lubos@nordicsemi.no>

MarekPieta committed:
applications: nrf_desktop: Reduce MCUboot FLASH size (serial recovery)

Decrease number of USBD endpoints in DTS to reduce FLASH consumption.
The MCUboot uses USB only for serial recovery over CDC ACM class.

Jira: NCSDK-22309

Signed-off-by: Marek Pieta <Marek.Pieta@nordicsemi.no>

MarekPieta committed:
modules: mcuboot: Disable Zephyr's MCUboot handling for child images

Change disables Zephyr's MCUboot handling in child images. The NCS
handles needed operations on its own. Change is needed to ensure
consistency for main application image and child images.

Jira: NCSDK-22347

Signed-off-by: Marek Pieta <Marek.Pieta@nordicsemi.no>

gregersrygg committed:
applications: asset_tracker_v2: Fix nrf7002ek_wifi

Workaround for enabling the external flash on nRF9160 DK when using
the nRF7002ek shield. Add external flash overlay to fix build failure.

Signed-off-by: Gregers Gram Rygg <gregers.gram.rygg@nordicsemi.no>

de-nordic commented:
@nordicjm Can you take a look?

nordicjm commented:
I do not think this is right and if the idea is to propagate up to the target ncs image, this will not work with sysbuild-ncs @tejlmand

MarekPieta committed:
applications: nrf_desktop: Align upload reset in DFU MCUmgr module

Change aligns the upload reset handling in the DFU MCUmgr application
module to newly introduced public API of MCUmgr.

Jira: NCSDK-21434

Signed-off-by: Marek Pieta <Marek.Pieta@nordicsemi.no>

gregersrygg committed:
boards: Clean up high drive mode for nrf9160dk and nrf9161dk

High drive mode is now enabled on SPI3 for the board definitions in
Zephyr. Clean up unnecessary enabling of high drive mode in overlays.

Signed-off-by: Gregers Gram Rygg <gregers.gram.rygg@nordicsemi.no>

MarekPieta commented:
Rebased

MarekPieta commented:
> I do not think this is right and if the idea is to propagate up to the target ncs image, this will not work with sysbuild-ncs @tejlmand

It seems that we cannot remove the `PARENT_SCOPE` here (the information would no longer be propagated to Zephyr build system so it would not work as expected). I also understand that the solution is temporary, has its drawbacks and may need to be improved later (I copied it from the bottom of the `CMakeLists.txt` file to make handling `mcuboot_secondary_app` variant image consistent with the main application image).

rlubos committed:
lib: nrf_modem_lib: trace_backends: uart: Align code to driver changes

nrfx UARTE driver renamed member of the configuration structure
and added new parameter to driver API.

Signed-off-by: Adam Wojasinski <adam.wojasinski@nordicsemi.no>
Signed-off-by: Robert Lubos <robert.lubos@nordicsemi.no>

rlubos commented:
@nordicjm @tejlmand Any feedback wrt above comment?

rlubos committed:
samples: peripheral: 802154_phy_test: Remove obsolete workaround

After the upmerge the sample no longer fit into FLASH on
nrf5340dk_nrf5340_cpunet. Remove an obsolete workaround from the
sample configuration to reduce sample FLASH footprint.

Signed-off-by: Robert Lubos <robert.lubos@nordicsemi.no>

rlubos committed:
manifest: Regular Zephyr upmerge

Update manifest to point to the latest Zephyr and Mcuboot revisions
after an upmerge.
Include fixes in other repositories.

Signed-off-by: Robert Lubos <robert.lubos@nordicsemi.no>

sebastiandraus committed:
zigbee: Enable NRF_STORE_REBOOT_TYPE_GPREGRET by default

This commit enables NRF_STORE_REBOOT_TYPE_GPREGRET Kconfig option
for all Zigbee samples so the reset reason is stored between device
reboot.

Signed-off-by: Sebastian Draus <sebastian.draus@nordicsemi.no>

tejlmand commented on modules/mcuboot/CMakeLists.txt:
```diff
@@ -28,6 +28,10 @@ endif()
   if (CONFIG_NCS_IS_VARIANT_IMAGE)
+    # NCS Handles everything regarding mcuboot, ensure Zephyr doesn't interfere.
+    # This is a temporary solution until Zephyr signing has been made more modular.
+    set(CONFIG_BOOTLOADER_MCUBOOT False PARENT_SCOPE)
```
this is not safe.

CONFIG_BOOTLOADER_MCUBOOT is a setting imported from the `.config`.
I acknowledge that it only impacts CMake configure time, and that CONFIG_BOOTLOADER_MCUBOOT is still set in autoconf.h and elsewhere in the system, such as python code loading the `.config`.

But in CMake, there can be customer code checking for this flag for other purposes than signing.

Has it been investigated why this happens for XIP images but not for other variant builds ?

MarekPieta replied:
Hi @tejlmand,

Yes, I am aware that the solution is not perfect and we should not manipulate Kconfig values inside CMake. I just make the behavior consistent with handling main application image - I actually used the same workaround (please take a look at the bottom of this CMakeLists.txt file, you will see the same hotfix applied for the main image):
```
# NCS Handles everything regarding mcuboot, ensure Zephyr doesn't interfere.
# This is a temporary solution until Zephyr signing has been made more modular.
set(CONFIG_BOOTLOADER_MCUBOOT False PARENT_SCOPE)
```

MarekPieta replied:
Could you provide a proper solution for the problem during the ongoing upmerge? Or maybe we could accept my temporary workaround and improve it later on (together with improving integration for the main application image)?

tejlmand replied:
I see that.
Not fond of that solution either, but at least you are consistent with what we are already doing.

Solution accepted, but please do a rebase.

MarekPieta replied:
I need to open a new PR to sdk-nrf main branch. Give me moment.

tejlmand reviewed (changes requested):
We should not start to manipulate Kconfig settings inside CMake.

Can result in unintended / unexpected errors, especially at NCS downstream users who are not aware of such manipulation.

MarekPieta commented:
Upmerge is done already, opened PR to main that introduces the change: https://github.com/nrfconnect/sdk-nrf/pull/11696.
Closing this one
